tests/microos/network_bonding.pm

```diff
@@ -27,11 +27,14 @@ sub run {
     my ($self) = @_;
     select_console 'root-console';
     # remove existing NM-managed connections (except loopback)
-    my @interfaces = grep { !/^lo/ } split('\n', script_output 'nmcli -t -f NAME con');
-    assert_script_run "nmcli con delete '$_'" for @interfaces;
+    my @connections_info = grep { !/^lo/ } split('\n', script_output 'nmcli -t -f DEVICE,UUID conn show --active');
+    my @devices = map { (split /:/)[0] } @connections_info;
+    my @connection_ids = map { (split /:/)[1] } @connections_info;
+    # can fail due to same ID pointing to different connections
+    script_run "nmcli con delete '$_'" for @connection_ids;
```

Comment:
IMO ignoring failures should be avoided if possible. Can you uniq `@connection_ids`?

Reply:
good point ! Thanks